File: Sieve.java
public class Sieve {
    
    private static void dgsieve(int max)
    {
	boolean[] e;

	e = new boolean[max];
	e[1] = true;
	for (int i = 2; i < max; i++)
	{
	    e[i] = true;
	} //for

	for(int j = 2; j < max; j++)
	{
	    if (e[j])
	    {
		for (int p = 2; (p*j) < max; p++)
		{
		    e[p *j] = false;
		} // for
	    } // if
	} // for
    } // dgsieve

    public static void main(String[] args)
    {
	long max;

	if (args.length != 1)
	{
	    System.out.println("Usage Sieve <max prime>");
	} else {
	    max = Long.parseLong(args[0], 10); 	    
	    dgsieve((int)max);
	} // if else
    } // main
} // Sieve`

File: Sieve.py
#!/usr/bin/python3
import sys #needed for command line arguments
from bitarray import bitarray

# Global variable to hold boolean array as a bitarray
e = bitarray()
   
def dgsieve(max):
    global e
    print(max)
    #Check ifit's an even value
    if max&1 != 1 :
        return(False)
    for i in range(0,max+1):
        e.append(True)
    for j in range(3,max,2):
        if (e[j]):
            for p in range(j+j,max+1,j):
                e[p] = False
    return e[max]

#Main code block
def main():
    print ("Damian's Sieve of Eratosthenes")
    l = len(sys.argv)
    print(l)
    if (l!=2):
        print("Usage: Sieve2.py <number>")
        quit()
    n = int(sys.argv[1])
    if (dgsieve(n)):
            print("Is Prime")
    print("End")

main()
